### PR TITLE
doc(scripts): command line and schematics doc generation

### PR DESCRIPTION
**Pull request by bcabanes**
Generate documentation from Nx apis for:
- builders;
- schematics;
- commands.

Run `yarn documentation` to trigger the generation.

### CODE REVIEW

File: scripts/documentation/builders.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+/**
+ * @WhatItDoes: Generates default documentation from the builders' schema.
+ *    We need to construct an Array of objects containing all the information
+ *    of the builders and their associates schema info. It should be easily
+ *    parsable in order to be used in a rendering process using template. This
+ *    in order to generate a markdown file for each available schematic.
+ */
+const buildersSourceDirectory = path.join(
+  path.resolve(__dirname),
```
Comment (vsavkin) [inline scripts/documentation/builders.ts]: why `path.resolve(__dirname)` instead of `__dirname`?

File: scripts/documentation/builders.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    return parseJsonSchemaToOptions(registry, builder.rawSchema)
+      .then(options => ({ ...builder, options }))
+      .catch(error =>
+        console.error(`Can't parse schema option of ${builder.name}:\n${error}`)
+      );
+  });
+}
+
+function generateTemplate(builder): { name: string; template: string } {
+  let template = dedent`
```
Comment (vsavkin) [inline scripts/documentation/builders.ts]: const?

**Review by vsavkin — APPROVED** [review]

File: scripts/documentation/builders.ts (hunk 2 above)
Comment (bcabanes) [reply scripts/documentation/builders.ts]: I can't use `const` because the template variable is modified, enhanced for each options, see above line 71.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
These docs look great 😄 

Can we add a check in CI that `yarn documentation` produced no changes?
This way we can ensure that documentation updates are in the same commit as changes to functionality.

**Pull request merged**